File: Easy/N meetings in one room/n-meetings-in-one-room.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution
{
    public:
    //Function to find the maximum number of meetings that can
    //be performed in a meeting room.
   int maxMeetings(int start[], int end[], int n)
{
    // Create a vector of pairs, where each pair represents the end time and start time of a meeting.
    pair<int, int> meetings[n];
    for (int i = 0; i < n; i++)
        meetings[i] = make_pair(end[i], start[i]);

    // Sort the vector of pairs based on the end times of meetings in ascending order.
    sort(meetings, meetings + n);

    int x = 0;  // Variable to store the end time of the last selected meeting.
    int ans = 0;  // Variable to store the count of selected meetings.

    // Iterate through the sorted meetings.
    for (int i = 0; i < n; i++) {
        // If the start time of the current meeting is greater than or equal to the end time of the last selected meeting,
        // then select the current meeting and update the end time variable.
        if (meetings[i].second > x) {
            ans++;
            x = meetings[i].first;
        }
    }

    // Return the count of selected meetings.
    return ans;
}
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        int start[n], end[n];
        for (int i = 0; i < n; i++) cin >> start[i];

        for (int i = 0; i < n; i++) cin >> end[i];

        Solution ob;
        int ans = ob.maxMeetings(start, end, n);
        cout << ans << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Easy/N meetings in one room/n-meetings-in-one-room.java
//{ Driver Code Starts
import java.io.*;
import java.util.*;
import java.lang.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());

        while (t-- > 0) {
            String inputLine[] = br.readLine().trim().split(" ");
            int n = Integer.parseInt(inputLine[0]);

            int start[] = new int[n];
            int end[] = new int[n];

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++)
                start[i] = Integer.parseInt(inputLine[i]);

            inputLine = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) 
                end[i] = Integer.parseInt(inputLine[i]);
                
            int ans = new Solution().maxMeetings(start, end, n);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends

class Sort {
    int s;
    int e;
    public Sort(int s, int e){
        this.s = s;
        this.e = e;
    }
}

class Solution {
    // Function to find the maximum number of meetings that can be performed in a meeting room.
    public static int maxMeetings(int start[], int end[], int n) {
        Sort[] sort = new Sort[n];
        for (int i = 0; i < n; i++) {
            sort[i] = new Sort(start[i], end[i]);
        }
        Arrays.sort(sort, (x, y) -> Integer.compare(x.e, y.e));
        
        int c = 1;
        int prev = sort[0].e;
        
        for (int i = 1; i < sort.length; i++) {
            if (sort[i].s > prev) {
                c++;
                prev = sort[i].e;
            }
        }
        return c;
    }
}
        
        
        
